catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java

```diff
@@ -106,32 +108,37 @@
                     + "\tComplex:   search --cql \"title like 'some text' AND modified before 2012-09-01T12:30:00Z\"")
     String cqlFilter = null;
 
-    private List<Metacard> failedMetacards = Collections
-            .synchronizedList(new ArrayList<Metacard>());
+    @Option(name = "--max", required = false, aliases = {
+            "-max"}, multiValued = false, description = "Option to specify a maximum amount of metacards to ingest.")
+    int maxMetacards;
+
+    private List<Metacard> failedMetacards =
+            Collections.synchronizedList(new ArrayList<>());
 
     abstract List<Metacard> query(CatalogFacade facade, int startIndex, Filter filter);
 
     /**
-     *
-     * @param queryFacade
-     *            - the CatalogFacade used for query
-     * @param ingestFacade
-     *            - the CatalogFacade used for ingest
-     * @param startIndex
-     *            - the start index of the query
-     * @param filter
-     *            - the filter to query with
+     * @param queryFacade  - the CatalogFacade used for query
+     * @param ingestFacade - the CatalogFacade used for ingest
+     * @param startIndex   - the start index of the query
+     * @param filter       - the filter to query with
      * @return - the number of successfully created metacards.
      */
     protected int queryAndIngest(CatalogFacade queryFacade, CatalogFacade ingestFacade,
             int startIndex, Filter filter) {
-        List<Metacard> queryMetacards = null;
+        List<Metacard> queryMetacards;
         queryMetacards = query(queryFacade, startIndex, filter);
 
         if (queryMetacards == null || queryMetacards.isEmpty()) {
             return 0;
         }
 
+        // If maxMetacards is set, restrict the amount of ingested metacards to the maxMetacards
+        if (maxMetacards > 0 && ((queryMetacards.size() + successCount.get()) > maxMetacards)) {
```

Comment:
Why can't we just set the pagesize of the query to maxMetacards when maxMetacards - successCount > batchSize.

Seems like if I decided I wanted to set a scheduled command to pull back at most 1 records from a remote site every 5 minutes, and I didn't set the batch size, I would end uprequesting up to 1000 records every 5 minutes and then dropping all but 1 on the ground